hazelcast/src/main/java/com/hazelcast/internal/config/override/ExternalConfigurationOverride.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.internal.config.override;
+
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.client.config.impl.YamlClientDomConfigProcessor;
+import com.hazelcast.config.Config;
+import com.hazelcast.internal.config.YamlMemberDomConfigProcessor;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+import static com.hazelcast.internal.config.override.PropertiesToNodeConverter.propsToNode;
+import static java.lang.String.format;
+
+/**
+ * A class used to process external configuration overrides coming from environment variables/system properties,
+ * and applying them over existing {@link Config}/{@link ClientConfig} configuration
+ */
+public class ExternalConfigurationOverride {
+
+    private static final ILogger LOGGER = Logger.getLogger(ExternalConfigurationOverride.class);
+
+    public Config overwriteMemberConfig(Config config) {
+        return overwrite(config, (provider, c) -> {
```

Comment:
The problem is that we need to have access to multiple providers to validate if there are no conflicts(in your example configs from various sources are applied individually) so either we repeat the code responsible for iteration and validation, or we pull it out somehow (what I tried).

The method needs to be generic since `Config` and `ClientConfig` don't have a common denominator so I needed to play around with generics to make it possible and inject a passing function representing a strategy for applying the config